src/tallies/tally.cpp

```diff
@@ -282,6 +282,21 @@ Tally::Tally(pugi::xml_node node)
         "Invalid estimator '{}' on tally {}", est, id_)};
     }
   }
+
+#ifdef LIBMESH
+  // ensure a tracklength tally isn't used with a libMesh filter
+  for (auto i : this->filters_) {
+    auto df = dynamic_cast<MeshFilter*>(model::tally_filters[i].get());
+    if (df) {
+      auto lm = dynamic_cast<LibMesh*>(model::meshes[df->mesh()].get());
+      if (lm && estimator_ == TallyEstimator::TRACKLENGTH) {
+        fatal_error("A tracklength estimator cannot be used with "
```

Comment:
It looks like there are a few different places that you check whether the estimator is TRACKLENGTH. Do we need this one here in addition to the check in `bins_crossed`?

Reply:
Good question. In my mind, they do serve different purposes. This check helps us by failing early when reading tallies from XML input. The other in `bins_crossed` provides some helpful output if a user is setting up tallies using the CAPI/C++API and manages to define a tally using a LibMesh filter and TL estimator -- in which case this check would be bypassed.

Reply:
gotcha, sounds good!